src/gcs.rs

```diff
@@ -640,23 +642,64 @@ mod tests {
     async fn get_private_object() {
         init();
 
+        // define the private object
         let private_bucket = test_bucket();
         let filename = "get_private_object.txt";
 
+        // generate test data of a few kB
+        let file_length = 4096;
+        let rand_string: String = thread_rng()
+            .sample_iter(&Alphanumeric)
+            .take(file_length)
+            .map(char::from)
+            .collect();
+        // upload the test data
+        let client = &new_client();
+        let mut object_cursor = create_object_with_client(client, &private_bucket, filename)
+            .await
+            .unwrap();
+        append_bytes_with_client(
+            client,
+            &mut object_cursor,
+            &Vec::from(rand_string.into_bytes()),
+        )
+        .await
+        .unwrap();
+        finalize_upload_with_client(client, &mut object_cursor)
+            .await
+            .unwrap();
+
+        // test getting the private object
         let url = format!(
             "https://www.googleapis.com/storage/v1/b/{}/o/{}",
             private_bucket, filename
         );
 
+        // get object metadata
         let object_url = Url::parse(&url).unwrap();
-        let object: Object = get_object(object_url).await.unwrap();
+        let object = get_object(object_url).await.unwrap();
         println!("Object has {} bytes", object.size);
+        assert!(object.size == file_length as u64);
```

Comment:
There are a lot of "file_length as u64" why not declare it as u64?

Reply:
Well, it's kind of a coinflip. I need it as a usize three times, and as a u64 three times. I switched it, but now there are three "as usize" casts. I think this is unavoidable... In fact before I committed this I went down a rabbit hole of why Rust does ints this way and learned about how seriously it takes portability and future-proofing.

I settled on usize because that's what I needed for comparisons to things like byte array length, and the u64 casts were mostly for args to gcsfuser functions. I liked having the casts in function calls rather than assertions. I'm good either way tho.